src/Common/SSHWrapper.cpp

```diff
@@ -119,19 +93,20 @@ bool SSHKey::isEqual(const SSHKey & other) const
 
 String SSHKey::signString(std::string_view input) const
 {
-    SSHString input_str(input);
-    ssh_string output = nullptr;
-    if (int rc = pki_sign_string(key, input_str.get(), &output); rc != SSH_OK)
-        throw Exception(ErrorCodes::LIBSSH_ERROR, "Error singing with ssh key");
-    SSHString output_str(output);
-    return output_str.toString();
+    char * signature = nullptr;
+    if (int rc = sshsig_sign(input.data(), input.size(), key, nullptr, "clickhouse", SSHSIG_DIGEST_SHA2_256, &signature); rc != SSH_OK)
+        throw Exception(ErrorCodes::LIBSSH_ERROR, "Error signing with ssh key");
+    std::unique_ptr<char, CStringDeleter> sig_ptr(signature);
```

Comment:
libssh doxygen says about the last parameter of `sshsig_sign`:

```
Pointer to store the allocated signature string in the
armored format. Must be freed with
ssh_string_free_char()
```

`CStringDeleter` uses `std::free` instead of `ssh_string_free_char`.